### PR TITLE
[cmake] Update to 3.17.2

### PR DESCRIPTION
**Pull request by JackBoosY**
Updated to 3.17.2 to solve the issues of decompressing the source.

Related: #10657 #10552 #10336 #6059 #10492 #5951 #9078 #8603.

curlpp regression will fix in #10535.

### CODE REVIEW

**Comment by Neumann-A** [comment]
If you update CMake could you also update ninja to 1.10 (#10369)?
This way we can finally include fortran ;)

**Comment by JackBoosY** [comment]
@Neumann-A I think we should put them in different PRs.

**Comment by Voskrese** [comment]
@JackBoosY 
merge forcibly, error somewhere in CI

**Comment by JackBoosY** [comment]
For `osg:x64-osx` regression and `replxx` regression, I cannot reproduce it on my machine.
For `drlibs` regression, this is because the pipeline machine cannot download the source code.

**Review by PhoebeHui — APPROVED** [review]
LGTM

**Comment by Voskrese** [comment]
those who install cmake together with msvc, how to be? maybe don't put a minimal version

**Comment by JackBoosY** [comment]
```
/Volumes/data/work/1/s/installed/x64-osx/include/armadillo_bits/include_superlu.hpp:95:12: fatal error: '/Volumes/data/work/1/s/installed/x64-osx/include/supermatrix.h' file not found
  #include ARMA_INCFILE_WRAP(ARMA_SLU_SUPERMATRIX_H)
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I canot repro this on my machine.

**Comment by Neumann-A** [comment]
CMake has been updated to 3.17.2 in the meantime

**Comment by Neumann-A** [comment]
@JackBoosY Any change this PR gets merged any time soon?

**Comment by Voskrese** [comment]
@ras0219-msft merge this PR

Thanks.

**Comment by JackBoosY** [comment]
@Neumann-A  @Voskrese I will update to 3.17.2 in this week.

**Comment by JackBoosY** [comment]
There are some errors in pipeline, @BillyONeal could you please help me to take a look?

Thanks.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -235,6 +234,7 @@ civetweb:arm64-windows      = skip
 civetweb:arm-uwp            = skip
 civetweb:x64-uwp            = skip
 clapack:x64-uwp=fail
+clblas:arm64-windows=fail
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:237]: Was this supposed to be clblast ?

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1030,6 +1017,7 @@ mathgl:x64-uwp=fail
 mathgl:x64-linux=ignore
 matio:x64-linux=fail
 matio:x64-osx=fail
+magnum:arm64-windows
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:1020]:
```suggestion
magnum:arm64-windows=fail
```

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -998,8 +985,8 @@ llgl:x64-uwp=fail
 llvm:arm-uwp=fail
 llvm:arm64-windows=fail
 llvm:x64-linux=fail
+llvm:x64-osx=fail
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:988]: Do we know why llvm broke?

**Review by BillyONeal — COMMENTED** [review]
Looks like CI failed due to the missing =fail

File: scripts/ci.baseline.txt (hunk 3 above)
Comment (JackBoosY) [reply scripts/ci.baseline.txt:988]: Same with x64-linux. See #10295.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (JackBoosY) [reply scripts/ci.baseline.txt:237]: No, that's `clblas`.

**Comment by Voskrese** [comment]
@ras0219-msft merge this PR

Thanks.

**Comment by JackBoosY** [comment]
llvm regression will be solved in #11268.

**Comment by JackBoosY** [comment]
Regression list:
```
- [ ] podofo:x64-linux
- [ ] nanodbc:x64-linux
- [ ] opencc:x64-linux
- [ ] nanogui:arm64-windows
- [ ] magnum:arm64-windows
- [ ] nettle:x64-windows
- [ ] llvm:x64-windows
- [ ] llvm:x64-windows-static
- [ ] llvm:x86-windows
```

**Comment by JackBoosY** [comment]
@strega-nil Build `nanodbc: x64-linux` succeeded on my machine, configuration information shows:
```
-- nanodbc build: ODBC on Unix - unixODBC
-- ODBC compile flags: -I/usr/include -DHAVE_UNISTD_H -DHAVE_PWD_H -DHAVE_SYS_TYPES_H -DHAVE_LONG_LONG -DSIZEOF_LONG_INT=8
-- ODBC link flags:
```
It looks like we need to install the odbc package on the pipeline machine.

**Comment by strega-nil** [comment]
Assigning @BillyONeal as the owner of CI

**Comment by BillyONeal** [comment]
@strega-nil I don't see any CI system edits here?

**Comment by BillyONeal** [comment]
Oh, I see, it's in the chat. @JackBoosY Can you add the changes necessary to do that to provision-image.ps1 ?

**Comment by Voskrese** [comment]
@JackBoosY 
everyone waiting for you ))

**Comment by JackBoosY** [comment]
- [ ] itk:x64-osx
```
[1621/1622] : && /Library/Developer/CommandLineTools/usr/bin/c++  -fPIC  -mtune=native -march=corei7 -Wall -Wcast-align -Wdisabled-optimization -Wextra -Wformat=2 -Winvalid-pch -Wno-format-nonliteral -Wpointer-arith -Wshadow -Wunused -Wwrite-strings -funit-at-a-time -Wno-strict-overflow -Wno-deprecated -Wno-invalid-offsetof -Wno-undefined-var-template -Woverloaded-virtual  -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names -fPIE -Xlinker -pie Modules/Core/TestKernel/src/CMakeFiles/itkTestDriver.dir/itkTestDriver.cxx.o  -o bin/itkTestDriver  lib/libITKIOBMP-5.0.a  lib/libITKIOGDCM-5.0.a  lib/libITKIOGIPL-5.0.a  lib/libITKIOJPEG-5.0.a  lib/libITKIOMeshBYU-5.0.a  lib/libITKIOMeshFreeSurfer-5.0.a  lib/libITKIOMeshGifti-5.0.a  lib/libITKIOMeshOBJ-5.0.a  lib/libITKIOMeshOFF-5.0.a  lib/libITKIOMeshVTK-5.0.a  lib/libITKIOMeta-5.0.a  lib/libITKIONIFTI-5.0.a  lib/libITKIONRRD-5.0.a  lib/libITKIOPNG-5.0.a  lib/libITKIOTIFF-5.0.a  lib/libITKIOVTK-5.0.a  lib/libITKCommon-5.0.a  lib/libITKIOImageBase-5.0.a  lib/libITKTestKernel-5.0.a  lib/libITKIOBMP-5.0.a  lib/libITKIOGDCM-5.0.a  lib/libitkgdcmMSFF-5.0.a  lib/libitkgdcmDICT-5.0.a  lib/libitkgdcmIOD-5.0.a  lib/libitkgdcmDSED-5.0.a  lib/libitkgdcmCommon-5.0.a  -framework  CoreFoundation  lib/libitkgdcmjpeg8-5.0.a  lib/libitkgdcmjpeg12-5.0.a  lib/libitkgdcmjpeg16-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libopenjp2.a  lib/libitkgdcmcharls-5.0.a  lib/libitkgdcmuuid-5.0.a  lib/libITKIOGIPL-5.0.a  lib/libITKIOJPEG-5.0.a  lib/libITKIOMeshBYU-5.0.a  lib/libITKIOMeshFreeSurfer-5.0.a  lib/libITKIOMeshGifti-5.0.a  lib/libITKgiftiio-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libexpat.a  lib/libITKIOMeshOBJ-5.0.a  lib/libITKIOMeshOFF-5.0.a  lib/libITKIOMeshVTK-5.0.a  lib/libITKIOMeshBase-5.0.a  lib/libITKQuadEdgeMesh-5.0.a  lib/libITKMesh-5.0.a  lib/libITKIOMeta-5.0.a  lib/libITKMetaIO-5.0.a  lib/libITKIONIFTI-5.0.a  lib/libITKniftiio-5.0.a  lib/libITKznz-5.0.a  lib/libITKTransform-5.0.a  lib/libITKIONRRD-5.0.a  lib/libITKNrrdIO-5.0.a  lib/libITKIOPNG-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libpng16d.a  lib/libITKIOTIFF-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libtiffd.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/liblzmad.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libz.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libjpeg.a  lib/libITKIOVTK-5.0.a  lib/libITKIOImageBase-5.0.a  lib/libITKCommon-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libdouble-conversion.a  lib/libitksys-5.0.a  lib/libITKVNLInstantiation-5.0.a  lib/libitkvnl_algo-5.0.a  lib/libitkvnl-5.0.a  lib/libitkv3p_netlib-5.0.a  lib/libitknetlib-5.0.a  lib/libitkvcl-5.0.a  -lm  -lm && :
FAILED: bin/itkTestDriver 
: && /Library/Developer/CommandLineTools/usr/bin/c++  -fPIC  -mtune=native -march=corei7 -Wall -Wcast-align -Wdisabled-optimization -Wextra -Wformat=2 -Winvalid-pch -Wno-format-nonliteral -Wpointer-arith -Wshadow -Wunused -Wwrite-strings -funit-at-a-time -Wno-strict-overflow -Wno-deprecated -Wno-invalid-offsetof -Wno-undefined-var-template -Woverloaded-virtual  -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names -fPIE -Xlinker -pie Modules/Core/TestKernel/src/CMakeFiles/itkTestDriver.dir/itkTestDriver.cxx.o  -o bin/itkTestDriver  lib/libITKIOBMP-5.0.a  lib/libITKIOGDCM-5.0.a  lib/libITKIOGIPL-5.0.a  lib/libITKIOJPEG-5.0.a  lib/libITKIOMeshBYU-5.0.a  lib/libITKIOMeshFreeSurfer-5.0.a  lib/libITKIOMeshGifti-5.0.a  lib/libITKIOMeshOBJ-5.0.a  lib/libITKIOMeshOFF-5.0.a  lib/libITKIOMeshVTK-5.0.a  lib/libITKIOMeta-5.0.a  lib/libITKIONIFTI-5.0.a  lib/libITKIONRRD-5.0.a  lib/libITKIOPNG-5.0.a  lib/libITKIOTIFF-5.0.a  lib/libITKIOVTK-5.0.a  lib/libITKCommon-5.0.a  lib/libITKIOImageBase-5.0.a  lib/libITKTestKernel-5.0.a  lib/libITKIOBMP-5.0.a  lib/libITKIOGDCM-5.0.a  lib/libitkgdcmMSFF-5.0.a  lib/libitkgdcmDICT-5.0.a  lib/libitkgdcmIOD-5.0.a  lib/libitkgdcmDSED-5.0.a  lib/libitkgdcmCommon-5.0.a  -framework  CoreFoundation  lib/libitkgdcmjpeg8-5.0.a  lib/libitkgdcmjpeg12-5.0.a  lib/libitkgdcmjpeg16-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libopenjp2.a  lib/libitkgdcmcharls-5.0.a  lib/libitkgdcmuuid-5.0.a  lib/libITKIOGIPL-5.0.a  lib/libITKIOJPEG-5.0.a  lib/libITKIOMeshBYU-5.0.a  lib/libITKIOMeshFreeSurfer-5.0.a  lib/libITKIOMeshGifti-5.0.a  lib/libITKgiftiio-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libexpat.a  lib/libITKIOMeshOBJ-5.0.a  lib/libITKIOMeshOFF-5.0.a  lib/libITKIOMeshVTK-5.0.a  lib/libITKIOMeshBase-5.0.a  lib/libITKQuadEdgeMesh-5.0.a  lib/libITKMesh-5.0.a  lib/libITKIOMeta-5.0.a  lib/libITKMetaIO-5.0.a  lib/libITKIONIFTI-5.0.a  lib/libITKniftiio-5.0.a  lib/libITKznz-5.0.a  lib/libITKTransform-5.0.a  lib/libITKIONRRD-5.0.a  lib/libITKNrrdIO-5.0.a  lib/libITKIOPNG-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libpng16d.a  lib/libITKIOTIFF-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libtiffd.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/liblzmad.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libz.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libjpeg.a  lib/libITKIOVTK-5.0.a  lib/libITKIOImageBase-5.0.a  lib/libITKCommon-5.0.a  /Volumes/data/work/1/s/installed/x64-osx/debug/lib/libdouble-conversion.a  lib/libitksys-5.0.a  lib/libITKVNLInstantiation-5.0.a  lib/libitkvnl_algo-5.0.a  lib/libitkvnl-5.0.a  lib/libitkv3p_netlib-5.0.a  lib/libitknetlib-5.0.a  lib/libitkvcl-5.0.a  -lm  -lm && :
Undefined symbols for architecture x86_64:
  "gdcm::Rescaler::SetMinMaxForPixelType(double, double)", referenced from:
      itk::GDCMImageIO::Write(void const*) in libITKIOGDCM-5.0.a(itkGDCMImageIO.cxx.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

- [ ] osg:x64-linux
```
/mnt/_work/1/s/buildtrees/osg/src/raph-3.6.4-c5448290ed/src/osgPlugins/RestHttpDevice/connection.cpp:43:29: error: ‘error’ is not a member of ‘asio::placeholders’
         asio::placeholders::error,
                             ^~~~~
```

**Comment by JackBoosY** [comment]
Temporarily disable `itk:x64-osx`, `osg:x64-linux`, `nanodbc:x64-linux` in hope of merging the PR as soon as possible.

@strega-nil @BillyONeal

**Comment by Voskrese** [comment]
@JackBoosY 
transfer the code to yourself, or the merge will be delayed for another 100 years

https://github.com/microsoft/vcpkg/pull/11414/files
`provision-image.ps1`

**Review by cbezault — APPROVED** [review]

**Pull request merged**